### PR TITLE
Api2 refactor settings

### PR DESCRIPTION
**Pull request by CarlosQ96**
Related to https://github.com/givepraise/praise/issues/732
1. Refactored PeriodSettings Schema
2. Added new roles for periodSettings.
3. Fixed circular dependencies on e2e tests when using Quantification, periods, settings and period settings modules.
4. Added Swagger
5. Changed periodSettings controller to use RequestContext.
6. Added EventLogService to PeriodSettings.
7. Delete export endpoint
8. Added guards for period settings endpoint.

### CODE REVIEW

**Comment by CarlosQ96** [comment]
Now that I fixed circular dependencies, Finishing E2E test for praise settings, will push this tomorrow morning.

**Comment by kristoferlund** [comment]
Great! Let me know when you want to review.

File: packages/api2/src/database/migrations/06_event_log_types.ts [hunk 1]
```diff
@@ -40,6 +41,11 @@ const eventLogTypes = [
...
     description: 'An action to quantify praise',
   },
+  {
+    key: EventLogTypeKey.PERIOD_SETTING,
+    label: 'Period Setting',
+    description: 'An action to period setting',
+  },
 ];
 
 const up = async (): Promise<void> => {
```
Comment (kristoferlund) [inline packages/api2/src/database/migrations/06_event_log_types.ts:51]: You cannot change migrations "retroactively". If they have already run they cannot be changed. I removed change, let's use SETTING type for period setting changes as well.

**Review by kristoferlund — APPROVED** [review]
Great @CarlosQ96! I made some small changes and fixes, please see the commits. Merging.

**Pull request merged**